doc/vxlan/Vxlan_hld.md

```diff
@@ -0,0 +1,338 @@
+|--------------------------|-----------------------------|
+| VNI                      | 8k                          |
+| Tunnel encaps            | 128k                        |
+| VMs                      | 512k                        |
+| VRFs                     | 128                         |
+
+# 2 Modules Design
+
+## 2.1 Config DB
+Following new tables will be added to Config DB
+### 2.1.1 VXLAN Table
+
+
+```
+VXLAN_TUNNEL:{{tunnel_name}} 
+    "src_ip": {{ip_address}} 
+    "dst_ip": {{ip_address}}
+
+VXLAN_TUNNEL_MAP|{{tunnel_name}}|{{tunnel_map}}
+    "vni_id": {{ vni_id}}
+    "vlan_id": {{ vlan_id }}
+```
+### 2.1.2 VNET Table
+```
+VNET:{{vnet_name}} 
+    "vxlan_tunnel": {{tunnel_name}}
+    "vxlanid": {{vni}} 
+    "peer_list": {{vnet_name_list}} 
+
+VNET_INTF|{{intf_name}}|{{prefix}}  
+    "vnet_name": {{vnet_name}} 
+```
+
+### 2.1.2 Schemas
+```
+; Defines schema for VXLAN Tunnel configuration attributes
+key                                   = VXLAN_TUNNEL:name             ; Vxlan tunnel configuration
+; field                               = value
+SRC_IP                                = ipv4                          ; Ipv4 source address, lpbk address
+DST_IP                                = ipv4                          ; Ipv4 source address, lpbk address
+
+;value annotations
+ipv4          = dec-octet "." dec-octet "." dec-octet "." dec-octet     
+dec-octet     = DIGIT                     ; 0-9  
+                  / %x31-39 DIGIT         ; 10-99  
+                  / "1" 2DIGIT            ; 100-199  
+                  / "2" %x30-34 DIGIT     ; 200-249
+		  
+```
+
+```
+; Defines schema for VXLAN Tunnel map configuration attributes
+key                                   = VXLAN_TUNNEL:tunnel_name:name ; Vxlan tunnel configuration
+; field                               = value
+VNI_ID                                = DIGITS                        ; 1 to 16 million values
+VLAN_ID                               = 1\*4DIGIT                     ; 1 to 4094 Vlan id
+```
+
+```
+; Defines schema for VNet configuration attributes
+key                                   = VNET:name                     ; Vnet name
+; field                               = value
+VXLAN_TUNNEL                          = tunnel_name                   ; refers to the Vxlan tunnel name
+VNI_ID                                = DIGITS                        ; 1 to 16 million VNI values
+PEER_LIST                             = \*vnet_name                   ; vnet names seperate by "," 
+                                                                             (empty indicates no peering)
+```
+
+```
+; Defines schema for VNet Interface configuration attributes
+key                                   = VNET_INTF:name:prefix         ; Vnet interface name with IP prefix
+; field                               = value
+VNET_NAME                             = vnet_name                     ; vnet name where the interface belongs to
```

Comment:
For the schemas defined here,  are they for appDB or configDB?

Reply:
better to specify this clearly.

Reply:
I noticed that VNI to VLAN map is added to config DB, while VNI to virtual router map is added to app DB.
Is there a reason for separating these two types of tunnel maps?

Reply:
Yes, the idea is to have VxlanOrch only deal with Tunnel configurations (Encap/Decap etc). The peering must be required only by VRF/RouterOrch.